lib/Transforms/MemoryBanking.cpp

```diff
@@ -88,6 +90,121 @@ MemRefType computeBankedMemRefType(MemRefType originalType,
+  size_t numElements = std::reduce(memShape.begin(), memShape.end(), 1,
+                                   std::multiplies<size_t>());
+  // `bankingFactor` number of flattened attributes that store the information
+  // in the original globalOp.
+  SmallVector<SmallVector<Attribute>> subBlocks;
+  subBlocks.resize(bankingFactor);
+
+  for (unsigned linIndex = 0; linIndex < numElements; ++linIndex) {
+    SmallVector<int64_t> ndIndex = decodeIndex(linIndex, memShape);
+    unsigned subBlockIndex = ndIndex[bankingDimension] % bankingFactor;
+    subBlocks[subBlockIndex].push_back(allAttrs[linIndex]);
+  }
+
+  return subBlocks;
+}
+
+// Handles the splitting of a GetGlobalOp into multiple banked memory and
+// creates new GetGlobalOp to represent each banked memory by slicing the data
+// in the original GetGlobalOp.
+SmallVector<Value, 4> handleGetGlobalOp(memref::GetGlobalOp getGlobalOp,
+                                        uint64_t bankingFactor,
+                                        unsigned bankingDimension,
+                                        MemRefType newMemRefType,
+                                        OpBuilder &builder) {
+  SmallVector<Value, 4> banks;
+  auto memTy = cast<MemRefType>(getGlobalOp.getType());
+  ArrayRef<int64_t> originalShape = memTy.getShape();
+  auto newShape =
+      SmallVector<int64_t>(originalShape.begin(), originalShape.end());
+  newShape[bankingDimension] = originalShape[bankingDimension] / bankingFactor;
+
+  auto *symbolTableOp = getGlobalOp->getParentWithTrait<OpTrait::SymbolTable>();
+  auto globalOp = dyn_cast_or_null<memref::GlobalOp>(
+      SymbolTable::lookupSymbolIn(symbolTableOp, getGlobalOp.getNameAttr()));
+  assert(globalOp && "The corresponding GlobalOp should exist in the module");
+  MemRefType globalOpTy = globalOp.getType();
+
+  auto cstAttr =
+      dyn_cast_or_null<DenseElementsAttr>(globalOp.getConstantInitValue());
+  auto attributes = cstAttr.getValues<Attribute>();
+  SmallVector<Attribute, 8> allAttrs(attributes.begin(), attributes.end());
+
+  auto subBlocks =
+      sliceSubBlock(allAttrs, originalShape, bankingDimension, bankingFactor);
+
+  // Initialize globalOp and getGlobalOp's insertion points. Since
+  // bankingFactor is guaranteed to be greater than zero as it would
+  // have early exited if not, the loop below will execute at least
+  // once. So it's safe to manipulate the insertion points here.
+  builder.setInsertionPointAfter(globalOp);
+  OpBuilder::InsertPoint globalOpsInsertPt = builder.saveInsertionPoint();
+  builder.setInsertionPointAfter(getGlobalOp);
+  OpBuilder::InsertPoint getGlobalOpsInsertPt = builder.saveInsertionPoint();
+
+  for (size_t bankCnt = 0; bankCnt < bankingFactor; ++bankCnt) {
+    // Prepare relevant information to create a new GlobalOp
+    auto newMemRefTy = MemRefType::get(newShape, globalOpTy.getElementType());
+    auto newTypeAttr = TypeAttr::get(newMemRefTy);
+    std::string newNameStr =
```

Comment:
Nit: It is clear from the type already (std::string), IMO we don't need the `Str` suffix here.
```suggestion
    std::string newName =
```

Reply:
sounds reasonable!

Plan to merge the patch once the CI has passed. Thanks again for the review :-)